src/compas/scene/sceneobject.py

```diff
@@ -26,11 +27,24 @@ class SceneObject(object):
     # add this to support the descriptor protocol vor Python versions below 3.6
     __metaclass__ = DescriptorProtocol
 
+    color = ColorDictAttribute()
+    pointcolor = ColorDictAttribute()
+    linecolor = ColorDictAttribute()
+    facecolor = ColorDictAttribute()
```

Comment:
i guess a geometry object has points, curves (or lines), and surfaces.
each of these have one default color (not a color dict).

data structures have vertices, edges, faces.
there we want to provide the option to set different colors for different components of the same type.
so vertexcolor, edgecolor, facecolor are color dicts instead.